### ISSUE TITLE
Questions about vocoder

### ISSUE THREAD

**Issue by josephwong14wkh** [mct10/RepCodec#11]
Thank you for your excellent work! 
I am going to train a speech tokenizer based on this paper, and there are two questions regarding to the pretrained vocoder:

1. While testing the pretrained vocoder without the -vc flag on VCTK samples, I noticed that the speaker in the generated audio (e.g., xxx_gen.wav) differs from the speaker in the input audio (e.g., xxx_gt.wav). Is it an expected behavior?

2. After tokenizing the audio waveform with RepCodec, I observed that the vocoder (speech synthesis model) receives only the token IDs, which are then mapped to vectors using another embedding table within the vocoder. Could you explain the rationale behind passing only token IDs instead of directly using the token vectors trained in RepCodec? (vector dimension in repcodec is 1024, while vector dimension in vocoder is 128, so i think the embedding tables should be different). Is the embedding table in vocoder trained from scratch? Are there specific advantages to this approach?

**Comment by mct10** [comment]
Hi, thanks for reaching out.
1. No, I think by default the synthesized audio should use the same speaker identity as the input audio. I double checked on my side, and I think they sound similar (xxx_gen.wav and xxx_gt.wav).
Can you double check if you are comparing the correct files? e.g., `p263_021_mic2_gt.wav` and `p263_021_mic2_gen.wav`. Please make sure you are not comparing files like `p263_021_mic2_0_gen.wav` (generated by `-vc`) where `0` is the speaker id.
2. We use the speech synthesis tool to evaluate the performance of RepCodec units, and we need to compare RepCodec units with baselines like k-means units. As in, we care more about the units rather than the embeddings. Speech tokenizers are all about units but not the embeddings, I think.
    - Yes, the vocoder's embedding table is trained from scratch.
    - No, I don't think there are any clear advantages (or disadvantages). We train this embedding table for all kinds of units, in order to have a fair comparison.

**Comment by josephwong14wkh** [comment]
Thank you for the quick response! 

`...Speech tokenizers are all about units but not the embeddings ...`
I get the idea of speech tokenizer eventually. Thanks!

And for my first questions, I have figured out the bugs, and now the speaker of the generated audio is the same as that of input audio! Thanks!

Btw I have another question regarding to the training loss, and here is my training loss:

![Image](https://github.com/user-attachments/assets/c00dd09a-11f3-45b4-98ce-7c0a06713709)
As you can see, in the beginning of the training, the codec loss and reconstruction loss suddenly increased significantly (e.g. reconstruction loss suddenly jumped from ~45 to ~115) and return back to normal quickly. In the middle of the training, there are still a spike in the loss, but not that significant compared to before. May i know have you encountered this problem before? Are these training losses normal?

**Comment by mct10** [comment]
The curves look alright. My experience is that the reconstruction loss almost always decreases, and vqloss increases for the first few hundreds/thousands of steps and then decreases. You can also take a look at [these figures](https://github.com/mct10/RepCodec/issues/6#issuecomment-2299746044).
I am not sure about the spikes. It might be related to different datasets or types of embeddings. Or maybe, if you modified the code to use fp16/amp instead of fp32 for training, it could be less stable. But anyway, I guess they do not matter that much as the trend looks good.

**Comment by josephwong14wkh** [comment]
Got it! Thank you!
